drivers/ec_host_cmd_periph/ec_host_cmd_periph_shi_ite.c

```diff
@@ -0,0 +1,504 @@
+	 * IT83XX_SPI_TXRXFAR = 0 in shi_ite_reset_rx_fifo.
+	 */
+	for (int i = 0; i < count; i += 4) {
+		/* Get data from master to buffer */
+		*(uint32_t *)(in_msg_addr + i) = IT83XX_SPI_RXFRDRB0;
+	}
+}
+
+static int shi_ite_host_request_expected_size(const struct ec_host_cmd_request_header *r)
+{
+	/* Check host request version */
+	if (r->prtcl_ver != EC_HOST_REQUEST_VERSION) {
+		return 0;
+	}
+
+	/* Reserved byte should be 0 */
+	if (r->reserved) {
+		return 0;
+	}
+
+	return sizeof(*r) + r->data_len;
+}
+
+/* Parse header for version of spi-protocol */
+static void shi_ite_parse_header(const struct device *dev)
+{
+	struct shi_it8xxx2_data *data = dev->data;
+	struct ec_host_cmd_request_header *r = (struct ec_host_cmd_request_header *)data->in_msg;
+
+	/* Store request data from Rx FIFO to in_msg buffer */
+	shi_ite_host_request_data(data->in_msg, sizeof(*r));
+
+	/* Protocol version 3 */
+	if (data->in_msg[0] == EC_HOST_REQUEST_VERSION) {
+		/* Check how big the packet should be */
+		data->in_msg_size = shi_ite_host_request_expected_size(r);
+
+		if (data->in_msg_size == 0 || data->in_msg_size > sizeof(data->in_msg)) {
+			shi_ite_bad_received_data(dev, data->in_msg_size);
+			return;
+		}
+
+		/* Store request data from Rx FIFO to in_msg buffer */
+		shi_ite_host_request_data(data->in_msg + sizeof(*r),
+					  data->in_msg_size - sizeof(*r));
+
+		k_sem_give(&data->handler_owns);
+	} else {
+		/* Invalid version number */
+		LOG_ERR("Invalid version number");
+		shi_ite_bad_received_data(dev, 1);
+	}
+}
+
+static void shi_ite_int_handler(const struct device *dev)
+{
+	struct shi_it8xxx2_data *data = dev->data;
+
+	if (data->shi_state == SHI_STATE_DISABLED) {
+		return;
+	}
+
+	/*
+	 * The status of SPI end detection interrupt bit is set, it means that host command parse
+	 * has been completed and AP has received the last byte which is EC_SHI_PAST_END from
+	 * EC responded data, then AP ended the transaction.
+	 */
+	if (IT83XX_SPI_ISR & IT83XX_SPI_ENDDETECTINT) {
+		/* Disable CPU access Rx FIFO to clock in data from AP again */
+		IT83XX_SPI_TXRXFAR = 0;
+		/* Ready to receive */
+		shi_ite_set_state(data, SHI_STATE_READY_TO_RECV);
+		/* CS# is deasserted, so write clear all slave status */
+		IT83XX_SPI_ISR = 0xff;
+	}
+
+	/*
+	 * The status of Rx valid length interrupt bit is set that indicates reached target count
+	 * (IT83XX_SPI_FTCB1R, IT83XX_SPI_FTCB0R) and the length field of the host requested data.
+	 */
+	if (IT83XX_SPI_RX_VLISR & IT83XX_SPI_RVLI) {
+		/* write clear slave status */
+		IT83XX_SPI_RX_VLISR = IT83XX_SPI_RVLI;
+		/* Move to processing state */
+		shi_ite_set_state(data, SHI_STATE_PROCESSING);
+		/* Parse header for version of spi-protocol */
+		shi_ite_parse_header(dev);
+	}
+}
+
+void shi_ite_cs_callback(const struct device *port, struct gpio_callback *cb, gpio_port_pins_t pins)
+{
+	struct shi_it8xxx2_data *data = CONTAINER_OF(cb, struct shi_it8xxx2_data, cs_cb);
+
```

Comment:
When this GPIO interrupt is fired, it means that SHI starts to transmit data. We should prevent the device entering the power policy, we can use `pm_policy_state_lock_get()`.

Reply:
Added these functions, thank you very much